submit-api/src/submit_api/services/submission/submission_creator_factory.py

```diff
@@ -150,8 +153,14 @@ def _create_submission(session, item_id, submitted_document_id, original_submiss
             submitted_document_id=submitted_document_id,
             major_version=major_version,
             minor_version=minor_version,
-            created_by=TokenInfo.get_id()
-
+            created_by=TokenInfo.get_id(),
+            root_submission_id=root_submission_id
         )
         session.add(submission)
+
+        # Set `root_submission_id` to its own ID if not provided
+        if submission.root_submission_id is None:
+            submission.root_submission_id = submission.id
+            session.commit()
```

Comment:
should we commit at this point or just flush.

Reply:
The session will of course commit at the end
![image](https://github.com/user-attachments/assets/d194fc86-033d-4b7e-bcb7-433cd0170901)